### PR TITLE
Shift to version 8.0.0 of taskcluster-lib-api

### PR DESCRIPTION
**Pull request by kritisingh1**
Upgrade taskcluster-lib-api to 8.0.0 for incorporating recent changes

### CODE REVIEW

**Comment by kritisingh1** [comment]
Sure. I am on it.

**Comment by kritisingh1** [comment]
@djmitche tests are now failing for a reason I am not fully sure of. Could you help a bit? Thanks.

**Comment by djmitche** [comment]
```
     TypeError: this._options.context.forEach is not a function
      at API.router (node_modules/taskcluster-lib-api/lib/api.js:1005:25)
```
So, it looks like something is trying to call `forEach` on `context`.  The path is to the compiled version of api.js in taskcluster-lib-api, but I only see one occurrence of the string `context.forEach` in that file, at https://github.com/taskcluster/taskcluster-lib-api/blob/master/src/api.js#L821.  `forEach` is an array method, so I'm guessing it's not an array in this case.  Why would that be so?

**Comment by kritisingh1** [comment]
@djmitche I have updated the changes. Please review. Sorry for delay

File: src/api.js [hunk 1]
```diff
@@ -33,6 +34,7 @@ api.declare({
   method:      'put',
   route:       '/secret/:name(*)',
   name:        'set',
+  context:     ['prop1', 'prop2'],
```
Comment (djmitche) [inline src/api.js]: What's this about?  API methods don't have a `context` property, and there's nothing named `prop1` or `prop2` in this service..

**Pull request merged**